### DISCUSSION TITLE
Is it possible to write empty history to file ?

### DISCUSSION THREAD

**Discussion by Jai-JAP** [akinomyoga/ble.sh#339] [Q&A] [answered]
I am using ble-sh latest git & I want to clear my shell history which is too large.

My env info:
```
GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu) [Ubuntu 22.04.2 LTS]
ble.sh, version 0.4.0-devel4+1afc616b (noarch) [git 2.34.1, GNU Make 4.3, GNU Awk 5.1.0, API: 3.0 (GNU MPFR 4.1.0, GNU MP 6.2.1)]
bash-completion, version 2.11 (hash:b42f5d6a7ad6d4921ec73838ba54a96d6bd30936, 77071 bytes) (noarch)
locale: LANG=en_IN
terminal: TERM=xterm-256color wcwidth=auto-auto/15.0-2+ri, vte:6800 (65;6800;1)
```

When using `ble/builtin/history`; `history -c` works but immediately running ` history -w` doesn't work whereas `history -w` does work, **(Notice the preceding space to skip saving history)** but saves `history -w` to history.
Running ` command history -w` works as intended (saves empty history), but its not convenient every-time.

So, my query is:
Whether we can save empty history to file using `ble/builtin/history` as simply as running ` history -w` after `history -c`

**Comment by akinomyoga** [comment, marked as answer]
Thanks. I added changes to `ble/builtin/history -w` in commit c78e5c9f137a6e0d6d58ef17601f32e582ca06c9. Could you update ble.sh by running `ble-update` in a ble.sh session and check the behavior?

----

**Edit**: It should be noted that the behavior of `ble/builtin/history` is intentionally changed in the initialization stage of Bash (in `.bashrc` or `.bash_profile`) because Bash's default behavior is sometimes inconsistent there. It is generally not recommended to use `history` builtins inside `.bashrc` or `.bash_profile`.

**Reply by Jai-JAP** [reply]
Thanks. Now it works perfectly as I wanted. 
Thanks for your quick response.
